insights/specs/datasources/system_user_dirs.py

```diff
@@ -0,0 +1,79 @@
+"""
+Custom datasource for checking non-root owned packaged directories
+
+Author: Florian Festi <ffesti@redhat.com>
+"""
+import json
+import stat
+
+from insights import datasource
+from insights.core.spec_factory import DatasourceProvider
+
+
+def get_shells():
+    with open('/etc/shells') as f:
+        return set((line.strip() for line in f))
+
+
+def get_users():
+    shells = get_shells()
+    users = set()
+
+    with open('/etc/passwd') as f:
+        for line in f:
+            entry = line.split(":")
+            if len(entry) < 7:
+                continue
+            shell = entry[6].strip()
+            if shell not in shells or shell in ('/sbin/nologin', '/bin/nologin'):
+                continue
+            users.add(entry[0])
+    users.discard("root")
+    return users
+
+
+def get_groups(users):
+    groups = set()
+
+    with open('/etc/group') as f:
+        for line in f:
+            entry = line.split(":")
+            if len(entry) < 4:
+                continue
+            if entry[0] in users:
+                groups.add(entry[0])  # group for an user of interest
+            for u in entry[3].split(","):
+                u = u.strip()
+                if u in users:
+                    groups.add(entry[0])  # add groups containing a user
+    return groups
+
+
+@datasource()
+def system_user_dirs(_broker):
+    try:
+        import rpm
+    except ImportError:
+        return ""
```

Comment:
It would be more suitable to raise a `SkipComponent` here, in case the `rpm` module can't be imported successfully, to skip the following steps, and hence the parser won't be triggered.

Reply:
Yes, I was thinking exactly about this one. Maybe it would be better to move this block to the top of the file?

Reply:
It sounds good to me.